src/main/java/org/jenkinsci/plugins/runselector/selectors/AbstractSpecificRunSelector.java

```diff
@@ -47,18 +61,31 @@
         if (context.getLastMatchBuild() != null) {
             return null;
         }
-        return getBuild(job, context);
+
+        String resolvedParameter = context.getEnvVars().expand(parameter);
+        if (resolvedParameter.startsWith("$")) {
+            context.logDebug("Unresolved variable {0}", resolvedParameter);
+            return null;
+        }
+
+        Run<?, ?> run = getBuild(job, resolvedParameter);
+
+        if (run == null) {
+            context.logDebug("No such build {0} in {1}", parameter, job.getFullName());
+            return null;
+        }
+
+        return run;
     }
-    
+
     /**
      * Override this method to implement {@link AbstractSpecificRunSelector}.
-     * 
-     * @param job       the job to pick a build from.
-     * @param context   context for the current execution of runselector.
+     *
+     * @param job               the job to pick a build from
+     * @param resolvedParameter the parameter on which the search is performed
      * @return the build to select
-     * @throws IOException if an error occurs while performing the operation.
-     * @throws InterruptedException if any thread interrupts the current thread.
+     * @throws IOException if an error occurs while performing the operation
      */
     @CheckForNull
-    public abstract Run<?, ?> getBuild(@Nonnull Job<?, ?> job, @Nonnull RunSelectorContext context) throws IOException, InterruptedException;
+    public abstract Run<?, ?> getBuild(@Nonnull Job<?, ?> job, @Nonnull String resolvedParameter) throws IOException;
```

Comment:
I don't think this is a good idea.
`AbstractSpecificRunSelector` may be used by other plugins that determines a build by its own way.